htdocs/product/stock/class/mouvementstock.class.php

```diff
@@ -145,6 +145,19 @@ public function _create($user, $fk_product, $entrepot_id, $qty, $type, $price =
 	    // phpcs:disable
 	    global $conf, $langs;
 
+		// $qty may be a string with more than 1 sign, like "+-4" or "--12.65" (when correct_stock_batch or correct_stock get negative values)
```

Comment:
The method _create is a low level method. Sanitizing of data should be done at a higher level (qty should already be a numeric value with the correct sign. It may be negative or positive on purpose).